core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/Chars.java

```diff
@@ -0,0 +1,30 @@
+package org.eclipse.rdf4j.rio.turtle;
+
+/**
+ * Utilities that were taken from the Character but changed to be better suited for parsing
+ * without additional garbage.
+ * @author Nikola Petrov nikola.petrov@ontotext.com
+ */
+class Chars {
+	/**
+	 * Appends the characters from codepoint into the string builder. This is the same
+	 * as Character#toChars but prevents the additional char array garbage.
+	 * @param dst the destination in which to append the characters
+	 * @param codePoint the codepoint to be appended
+	 */
+	public static void append(StringBuilder dst, int codePoint) {
```

Comment:
Unfortunately, it is not quite that simple. OpenJDK source code is licensed under GPL, which means we can't reuse it in our own code (unless we then distribute the entire RDF4J framework under GPL license as well - which is not something we want to do :)).

Given that this is such a small code snippet though, and a really quite basic algorithm, I'll doublecheck with the Eclipse legal team if we can get away with this somehow.